### PR TITLE
feat(connectivity_plus): vpn support for android and linux

### LINKED ISSUE #1166: [Bug]: on Linux an active VPN connection yields ConnectivityResult.mobile
### Platform

Ubuntu 22.04

### Plugin

connectivity_plus

### Version

2.3.8

### Flutter SDK

3.3.3

### Steps to reproduce

1. Connect to a VPN
2. Run the `connectivity_plus` example app
3. It shows "Connection Status: `ConnectivityResult.mobile`"

The problem is due to the fact that `ConnectivityResult.mobile` is returned by default if the `NetworkManagerClient`'s `primaryConnectionType` is not one of "wireless", "ethernet" or "bluetooth" (it reads "vpn" in my case).

### Code Sample

_No response_

### Logs

```shell
[  +52 ms] executing: uname -m
[  +26 ms] Exit code 0 from: uname -m
[        ] x86_64
[   +6 ms] executing: [/home/loose/flutter/] git -c log.showSignature=false log -n 1 --pretty=format:%H
[  +10 ms] Exit code 0 from: git -c log.showSignature=false log -n 1 --pretty=format:%H
[        ] 18a827f3933c19f51862dde3fa472197683249d6
[   +1 ms] executing: [/home/loose/flutter/] git tag --points-at 18a827f3933c19f51862dde3fa472197683249d6
[  +26 ms] Exit code 0 from: git tag --points-at 18a827f3933c19f51862dde3fa472197683249d6
[        ] 3.3.3
[  +84 ms] executing: [/home/loose/flutter/] git rev-parse --abbrev-ref --symbolic @{upstream}
[   +7 ms] Exit code 0 from: git rev-parse --abbrev-ref --symbolic @{upstream}
[        ] origin/stable
[        ] executing: [/home/loose/flutter/] git ls-remote --get-url origin
[   +7 ms] Exit code 0 from: git ls-remote --get-url origin
[        ] https://github.com/flutter/flutter.git
[  +46 ms] executing: [/home/loose/flutter/] git rev-parse --abbrev-ref HEAD
[   +8 ms] Exit code 0 from: git rev-parse --abbrev-ref HEAD
[        ] stable
[  +64 ms] Artifact Instance of 'AndroidGenSnapshotArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterWebSdk' is not required, skipping update.
[   +4 ms] Artifact Instance of 'WindowsEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required, skipping update.
[  +55 ms] Artifact Instance of 'AndroidGenSnapshotArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterWebSdk' is not required, skipping update.
[   +1 ms] Artifact Instance of 'WindowsEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping update.
[   +2 ms] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required, skipping update.
[ +109 ms] Skipping pub get: version match.
[  +93 ms] Found plugin integration_test at /home/loose/flutter/packages/integration_test/
[  +15 ms] Found plugin connectivity_plus at /home/loose/projects/random/plus_plugins/packages/connectivity_plus/connectivity_plus/
[   +3 ms] Found plugin connectivity_plus_linux at /home/loose/projects/random/plus_plugins/packages/connectivity_plus/connectivity_plus_linux/
[   +2 ms] Found plugin connectivity_plus_macos at /home/loose/projects/random/plus_plugins/packages/connectivity_plus/connectivity_plus_macos/
[   +2 ms] Found plugin connectivity_plus_web at /home/loose/projects/random/plus_plugins/packages/connectivity_plus/connectivity_plus_web/
[   +1 ms] Found plugin connectivity_plus_windows at /home/loose/projects/random/plus_plugins/packages/connectivity_plus/connectivity_plus_windows/
[  +78 ms] Found plugin integration_test at /home/loose/flutter/packages/integration_test/
[   +1 ms] Found plugin connectivity_plus at /home/loose/projects/random/plus_plugins/packages/connectivity_plus/connectivity_plus/
[   +2 ms] Found plugin connectivity_plus_linux at /home/loose/projects/random/plus_plugins/packages/connectivity_plus/connectivity_plus_linux/
[        ] Found plugin connectivity_plus_macos at /home/loose/projects/random/plus_plugins/packages/connectivity_plus/connectivity_plus_macos/
[   +1 ms] Found plugin connectivity_plus_web at /home/loose/projects/random/plus_plugins/packages/connectivity_plus/connectivity_plus_web/
[        ] Found plugin connectivity_plus_windows at /home/loose/projects/random/plus_plugins/packages/connectivity_plus/connectivity_plus_windows/
[  +70 ms] Found plugin integration_test at /home/loose/flutter/packages/integration_test/
[   +1 ms] Found plugin connectivity_plus at /home/loose/projects/random/plus_plugins/packages/connectivity_plus/connectivity_plus/
[   +3 ms] Found plugin connectivity_plus_linux at /home/loose/projects/random/plus_plugins/packages/connectivity_plus/connectivity_plus_linux/
[   +1 ms] Found plugin connectivity_plus_macos at /home/loose/projects/random/plus_plugins/packages/connectivity_plus/connectivity_plus_macos/
[   +3 ms] Found plugin connectivity_plus_web at /home/loose/projects/random/plus_plugins/packages/connectivity_plus/connectivity_plus_web/
[        ] Found plugin connectivity_plus_windows at /home/loose/projects/random/plus_plugins/packages/connectivity_plus/connectivity_plus_windows/
[  +58 ms] Generating
/home/loose/projects/random/plus_plugins/packages/connectivity_plus/connectivity_plus/example/android/app/src/main/java/io/flutter/plugins/GeneratedPluginRegistrant.java
[ +109 ms] Initializing file store
[  +13 ms] Skipping target: gen_localizations
[   +6 ms] gen_dart_plugin_registrant: Starting due to {InvalidatedReasonKind.inputChanged: The following inputs have updated contents:
/home/loose/projects/random/plus_plugins/packages/connectivity_plus/connectivity_plus/example/.dart_tool/package_config_subset,/home/loose/projects/random/plus_plugins/pac
kages/connectivity_plus/connectivity_plus/example/.dart_tool/flutter_build/dart_plugin_registrant.dart}
[  +33 ms] Found plugin integration_test at /home/loose/flutter/packages/integration_test/
[        ] Found plugin connectivity_plus at /home/loose/projects/random/plus_plugins/packages/connectivity_plus/connectivity_plus/
[   +2 ms] Found plugin connectivity_plus_linux at /home/loose/projects/random/plus_plugins/packages/connectivity_plus/connectivity_plus_linux/
[        ] Found plugin connectivity_plus_macos at /home/loose/projects/random/plus_plugins/packages/connectivity_plus/connectivity_plus_macos/
[   +2 ms] Found plugin connectivity_plus_web at /home/loose/projects/random/plus_plugins/packages/connectivity_plus/connectivity_plus_web/
[   +1 ms] Found plugin connectivity_plus_windows at /home/loose/projects/random/plus_plugins/packages/connectivity_plus/connectivity_plus_windows/
[   +6 ms] gen_dart_plugin_registrant: Complete
[   +3 ms] Skipping target: _composite
[   +2 ms] complete
[   +7 ms] Launching lib/main.dart on Linux in debug mode...
[   +4 ms] /home/loose/flutter/bin/cache/dart-sdk/bin/dart --disable-dart-dev /home/loose/flutter/bin/cache/dart-sdk/bin/snapshots/frontend_server.dart.snapshot --sdk-root
/home/loose/flutter/bin/cache/artifacts/engine/common/flutter_patched_sdk/ --incremental --target=flutter --debugger-module-names --experimental-emit-debug-metadata
-DFLUTTER_WEB_AUTO_DETECT=true --output-dill /tmp/flutter_tools.SIWOKA/flutter_tool.KXGKSO/app.dill --packages
/home/loose/projects/random/plus_plugins/packages/connectivity_plus/connectivity_plus/example/.dart_tool/package_config.json -Ddart.vm.profile=false
-Ddart.vm.product=false --enable-asserts --track-widget-creation --filesystem-scheme org-dartlang-root --initialize-from-dill
build/c075001b96339384a97db4862b8ab8db.cache.dill.track.dill --source
/home/loose/projects/random/plus_plugins/packages/connectivity_plus/connectivity_plus/example/.dart_tool/flutter_build/dart_plugin_registrant.dart --source
package:flutter/src/dart_plugin_registrant.dart
-Dflutter.dart_plugin_registrant=file:///home/loose/projects/random/plus_plugins/packages/connectivity_plus/connectivity_plus/example/.dart_tool/flutter_build/dart_plugin_
registrant.dart --verbosity=error --enable-experiment=alternative-invalidation-strategy
[  +30 ms] Building Linux application...
[  +33 ms] <- compile package:connectivity_plus_example/main.dart
[   +4 ms] executing: [build/linux/x64/debug/] cmake -G Ninja -DCMAKE_BUILD_TYPE=Debug -DFLUTTER_TARGET_PLATFORM=linux-x64
/home/loose/projects/random/plus_plugins/packages/connectivity_plus/connectivity_plus/example/linux
[  +45 ms] -- Configuring done
[   +3 ms] -- Generating done
[  +13 ms] -- Build files have been written to: /home/loose/projects/random/plus_plugins/packages/connectivity_plus/connectivity_plus/example/build/linux/x64/debug
[   +9 ms] executing: ninja -C build/linux/x64/debug install
[  +56 ms] ninja: Entering directory `build/linux/x64/debug'
[+2055 ms] [1/5] Generating /home/loose/projects/random/plus_plugins/packages/connectivity_plus/connectivity_plus/example/linux/flutter/ephemeral/libflutter_linux_gtk.so,
/home/loose/projects/random/plus_plugins/packages/connectivity_plus/connectivity_plus/example/linux/flutter/ephemeral/flutter_linux/fl_basic_message_channel.h,
/home/loose/projects/random/plus_plugins/packages/connectivity_plus/connectivity_plus/example/linux/flutter/ephemeral/flutter_linux/fl_binary_codec.h,
/home/loose/projects/random/plus_plugins/packages/connectivity_plus/connectivity_plus/example/linux/flutter/ephemeral/flutter_linux/fl_binary_messenger.h,
/home/loose/projects/random/plus_plugins/packages/connectivity_plus/connectivity_plus/example/linux/flutter/ephemeral/flutter_linux/fl_dart_project.h,
/home/loose/projects/random/plus_plugins/packages/connectivity_plus/connectivity_plus/example/linux/flutter/ephemeral/flutter_linux/fl_engine.h,
/home/loose/projects/random/plus_plugins/packages/connectivity_plus/connectivity_plus/example/linux/flutter/ephemeral/flutter_linux/fl_json_message_codec.h,
/home/loose/projects/random/plus_plugins/packages/connectivity_plus/connectivity_plus/example/linux/flutter/ephemeral/flutter_linux/fl_json_method_codec.h,
/home/loose/projects/random/plus_plugins/packages/connectivity_plus/connectivity_plus/example/linux/flutter/ephemeral/flutter_linux/fl_message_codec.h,
/home/loose/projects/random/plus_plugins/packages/connectivity_plus/connectivity_plus/example/linux/flutter/ephemeral/flutter_linux/fl_method_call.h,
/home/loose/projects/random/plus_plugins/packages/connectivity_plus/connectivity_plus/example/linux/flutter/ephemeral/flutter_linux/fl_method_channel.h,
/home/loose/projects/random/plus_plugins/packages/connectivity_plus/connectivity_plus/example/linux/flutter/ephemeral/flutter_linux/fl_method_codec.h,
/home/loose/projects/random/plus_plugins/packages/connectivity_plus/connectivity_plus/example/linux/flutter/ephemeral/flutter_linux/fl_method_response.h,
/home/loose/projects/random/plus_plugins/packages/connectivity_plus/connectivity_plus/example/linux/flutter/ephemeral/flutter_linux/fl_plugin_registrar.h,
/home/loose/projects/random/plus_plugins/packages/connectivity_plus/connectivity_plus/example/linux/flutter/ephemeral/flutter_linux/fl_plugin_registry.h,
/home/loose/projects/random/plus_plugins/packages/connectivity_plus/connectivity_plus/example/linux/flutter/ephemeral/flutter_linux/fl_standard_message_codec.h,
/home/loose/projects/random/plus_plugins/packages/connectivity_plus/connectivity_plus/example/linux/flutter/ephemeral/flutter_linux/fl_standard_method_codec.h,
/home/loose/projects/random/plus_plugins/packages/connectivity_plus/connectivity_plus/example/linux/flutter/ephemeral/flutter_linux/fl_string_codec.h,
/home/loose/projects/random/plus_plugins/packages/connectivity_plus/connectivity_plus/example/linux/flutter/ephemeral/flutter_linux/fl_value.h,
/home/loose/projects/random/plus_plugins/packages/connectivity_plus/connectivity_plus/example/linux/flutter/ephemeral/flutter_linux/fl_view.h,
/home/loose/projects/random/plus_plugins/packages/connectivity_plus/connectivity_plus/example/linux/flutter/ephemeral/flutter_linux/flutter_linux.h, _phony_
[   +3 ms] [  +52 ms] executing: uname -m
[        ] [  +29 ms] Exit code 0 from: uname -m
[        ] [        ] x86_64
[        ] [   +4 ms] executing: [/home/loose/flutter/] git -c log.showSignature=false log -n 1 --pretty=format:%H
[        ] [   +7 ms] Exit code 0 from: git -c log.showSignature=false log -n 1 --pretty=format:%H
[        ] [        ] 18a827f3933c19f51862dde3fa472197683249d6
[        ] [   +1 ms] executing: [/home/loose/flutter/] git tag --points-at 18a827f3933c19f51862dde3fa472197683249d6
[        ] [  +24 ms] Exit code 0 from: git tag --points-at 18a827f3933c19f51862dde3fa472197683249d6
[        ] [        ] 3.3.3
[        ] [  +57 ms] executing: [/home/loose/flutter/] git rev-parse --abbrev-ref HEAD
[        ] [   +6 ms] Exit code 0 from: git rev-parse --abbrev-ref HEAD
[        ] [        ] stable
[        ] [  +38 ms] Artifact Instance of 'AndroidGenSnapshotArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'FlutterWebSdk' is not required, skipping update.
[        ] [   +2 ms] Artifact Instance of 'WindowsEngineArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'LinuxEngineArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required, skipping update.
[        ] [  +65 ms] executing: [/home/loose/flutter/] git rev-parse --abbrev-ref --symbolic @{upstream}
[        ] [   +7 ms] Exit code 0 from: git rev-parse --abbrev-ref --symbolic @{upstream}
[        ] [        ] origin/stable
[        ] [        ] executing: [/home/loose/flutter/] git ls-remote --get-url origin
[        ] [   +7 ms] Exit code 0 from: git ls-remote --get-url origin
[        ] [        ] https://github.com/flutter/flutter.git
[        ] [   +1 ms] Artifact Instance of 'MaterialFonts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'GradleWrapper' is not required, skipping update.
[        ] [        ] Artifact Instance of 'AndroidGenSnapshotArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'FlutterWebSdk' is not required, skipping update.
[        ] [        ] Artifact Instance of 'FlutterSdk' is not required, skipping update.
[        ] [        ] Artifact Instance of 'WindowsEngineArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping update.
[        ] [   +1 ms] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required, skipping update.
[        ] [        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'FontSubsetArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'PubDependencies' is not required, skipping update.
[        ] [  +31 ms] Initializing file store
[        ] [   +4 ms] Done initializing file store
[        ] [  +32 ms] Skipping target: gen_localizations
[        ] [   +9 ms] Skipping target: gen_dart_plugin_registrant
[        ] [ +359 ms] Skipping target: unpack_linux
[        ] [ +277 ms] Skipping target: kernel_snapshot
[        ] [ +173 ms] Skipping target: debug_bundle_linux-x64_assets
[        ] [   +1 ms] Persisting file store
[        ] [   +9 ms] Done persisting file store
[        ] [   +4 ms] build succeeded.
[        ] [   +6 ms] "flutter assemble" took 1,009ms.
[        ] [ +262 ms] ensureAnalyticsSent: 259ms
[        ] [   +1 ms] Running shutdown hooks
[        ] [        ] Shutdown hooks complete
[        ] [   +1 ms] exiting with code 0
[ +459 ms] [2/5] Building CXX object CMakeFiles/example.dir/flutter/generated_plugin_registrant.cc.o
[  +13 ms] [3/5] Building CXX object CMakeFiles/example.dir/my_application.cc.o
[ +195 ms] [4/5] Linking CXX executable intermediates_do_not_run/example
[        ] [4/5] Install the project...
[   +6 ms] -- Install configuration: "Debug"
[   +9 ms] -- Installing: /home/loose/projects/random/plus_plugins/packages/connectivity_plus/connectivity_plus/example/build/linux/x64/debug/bundle/example
[        ] -- Set runtime path of "/home/loose/projects/random/plus_plugins/packages/connectivity_plus/connectivity_plus/example/build/linux/x64/debug/bundle/example" to
"$ORIGIN/lib"
[        ] -- Installing: /home/loose/projects/random/plus_plugins/packages/connectivity_plus/connectivity_plus/example/build/linux/x64/debug/bundle/data/icudtl.dat
[   +1 ms] -- Installing:
/home/loose/projects/random/plus_plugins/packages/connectivity_plus/connectivity_plus/example/build/linux/x64/debug/bundle/lib/libflutter_linux_gtk.so
[  +51 ms] -- Installing: /home/loose/projects/random/plus_plugins/packages/connectivity_plus/connectivity_plus/example/build/linux/x64/debug/bundle/data/flutter_assets
[        ] -- Installing:
/home/loose/projects/random/plus_plugins/packages/connectivity_plus/connectivity_plus/example/build/linux/x64/debug/bundle/data/flutter_assets/fonts
[        ] -- Installing:
/home/loose/projects/random/plus_plugins/packages/connectivity_plus/connectivity_plus/example/build/linux/x64/debug/bundle/data/flutter_assets/fonts/MaterialIcons-Regular.
otf
[   +2 ms] -- Installing:
/home/loose/projects/random/plus_plugins/packages/connectivity_plus/connectivity_plus/example/build/linux/x64/debug/bundle/data/flutter_assets/shaders
[        ] -- Installing:
/home/loose/projects/random/plus_plugins/packages/connectivity_plus/connectivity_plus/example/build/linux/x64/debug/bundle/data/flutter_assets/shaders/ink_sparkle.frag
[        ] -- Installing:
/home/loose/projects/random/plus_plugins/packages/connectivity_plus/connectivity_plus/example/build/linux/x64/debug/bundle/data/flutter_assets/NOTICES.Z
[        ] -- Installing:
/home/loose/projects/random/plus_plugins/packages/connectivity_plus/connectivity_plus/example/build/linux/x64/debug/bundle/data/flutter_assets/FontManifest.json
[        ] -- Installing:
/home/loose/projects/random/plus_plugins/packages/connectivity_plus/connectivity_plus/example/build/linux/x64/debug/bundle/data/flutter_assets/kernel_blob.bin
[  +37 ms] -- Installing:
/home/loose/projects/random/plus_plugins/packages/connectivity_plus/connectivity_plus/example/build/linux/x64/debug/bundle/data/flutter_assets/version.json
[        ] -- Installing:
/home/loose/projects/random/plus_plugins/packages/connectivity_plus/connectivity_plus/example/build/linux/x64/debug/bundle/data/flutter_assets/AssetManifest.json
[  +22 ms] Building Linux application... (completed in 3.0s)
[ +294 ms] Observatory URL on device: http://127.0.0.1:35041/3i_9DzUIFLo=/
[   +3 ms] Caching compiled dill
[  +65 ms] Connecting to service protocol: http://127.0.0.1:35041/3i_9DzUIFLo=/
[ +146 ms] Launching a Dart Developer Service (DDS) instance at http://127.0.0.1:0, connecting to VM service at http://127.0.0.1:35041/3i_9DzUIFLo=/.
[  +95 ms] DDS is listening at http://127.0.0.1:40971/eOo5CbBpdKc=/.
[  +53 ms] Successfully connected to service protocol: http://127.0.0.1:35041/3i_9DzUIFLo=/
[  +45 ms] DevFS: Creating new filesystem on the device (null)
[  +20 ms] DevFS: Created new filesystem on the device (file:///tmp/exampleBKRAAM/example/)
[   +2 ms] Updating assets
[ +117 ms] Syncing files to device Linux...
[   +3 ms] Compiling dart to kernel with 0 updated files
[   +1 ms] Processing bundle.
[   +2 ms] <- recompile package:connectivity_plus_example/main.dart c7636d7a-1a57-4b77-97c4-296d6dfe051c
[        ] <- c7636d7a-1a57-4b77-97c4-296d6dfe051c
[   +2 ms] Bundle processing done.
[ +119 ms] Updating files.
[        ] DevFS: Sync finished
[   +1 ms] Syncing files to device Linux... (completed in 130ms)
[        ] Synced 0.0MB.
[   +1 ms] <- accept
[   +2 ms] Connected to _flutterView/0x55addcfef3b0.
[   +2 ms] Flutter run key commands.
[   +1 ms] r Hot reload. 🔥🔥🔥
[   +1 ms] R Hot restart.
[        ] h List all available interactive commands.
[        ] d Detach (terminate "flutter run" but leave application running).
[        ] c Clear the screen
[        ] q Quit (terminate the application on the device).
[        ] 💪 Running with sound null safety 💪
[   +1 ms] An Observatory debugger and profiler on Linux is available at: http://127.0.0.1:40971/eOo5CbBpdKc=/
[ +266 ms] The Flutter DevTools debugger and profiler on Linux is available at: http://127.0.0.1:9101?uri=http://127.0.0.1:40971/eOo5CbBpdKc=/
[+4492 ms] Service protocol connection closed.
[        ] Lost connection to device.
[        ] DevFS: Deleting filesystem on the device (file:///tmp/exampleBKRAAM/example/)
[   +6 ms] (example:24865): GLib-GObject-WARNING **: 13:25:24.049: ../../../gobject/gsignal.c:2731: instance '0x55addc7731b0' has no handler with id '246'
[ +246 ms] Ignored error while cleaning up DevFS: TimeoutException after 0:00:00.250000: Future not completed
[   +3 ms] "flutter run" took 9,819ms.
[  +75 ms] ensureAnalyticsSent: 72ms
[   +1 ms] Running shutdown hooks
[        ] Shutdown hooks complete
[   +1 ms] exiting with code 0
```

### Flutter Doctor

```shell
[✓] Flutter (Channel stable, 3.3.3, on Ubuntu 22.04.1 LTS 5.15.0-48-generic, locale en_US.UTF-8)
    • Flutter version 3.3.3 on channel stable at /home/loose/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 18a827f393 (7 days ago), 2022-09-28 10:03:14 -0700
    • Engine revision 5c984c26eb
    • Dart version 2.18.2
    • DevTools version 2.15.0

[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.0)
    • Android SDK at /home/loose/Android/Sdk/
    • Platform android-33, build-tools 33.0.0
    • Java binary at: /snap/android-studio/123/android-studio/jre/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)
    • All Android licenses accepted.

[✓] Chrome - develop for the web
    • CHROME_EXECUTABLE = /snap/bin/chromium

[✓] Linux toolchain - develop for Linux desktop
    • Ubuntu clang version 14.0.0-1ubuntu1
    • cmake version 3.22.1
    • ninja version 1.10.1
    • pkg-config version 0.29.2

[✓] Android Studio (version 2021.2)
    • Android Studio at /snap/android-studio/123/android-studio
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)

[✓] Connected device (2 available)
    • Linux (desktop) • linux  • linux-x64      • Ubuntu 22.04.1 LTS 5.15.0-48-generic
    • Chrome (web)    • chrome • web-javascript • Chromium 106.0.5249.91 snap

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!
```

### Checklist before submitting a bug

- [X] I Google'd a solution and I couldn't find it
- [X] I searched on StackOverflow for a solution and I couldn't find it
- [X] I read the README.md file of the plugin
- [X] I'm using the latest version of the plugin
- [X] All dependencies are up to date with `flutter pub upgrade`
- [X] I did a `flutter clean`
- [X] I tried running the example project

### PR DESCRIPTION
**Pull request by werainkhatri**
## Description

adds support for vpn as `ConnectivityResult` to android and linux.

for android - added it as there is a parameter for it, but haven't tested it on device as i don't have a VPN.
for linux - again, added it (taking the word of the issue author) but can't test it on device.
for macos / ios - i suppose they don't support knowing if a vpn is active as i wasn't able to find any doc supporting it. the APIs currently use don't have such a parameter.
for windows, web - tried searching for something, but not an expert so will leave it to them / a feature request to come up.

## Related Issues

Fixes #1166

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`).
This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I **haven't** updated the version in `pubspec.yaml` and `CHANGELOG.md`.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [ ] The analyzer (`flutter analyze`) does not report any problems on my PR. (`melos analyze` fails locally but not related to my PR).
- [x] I read and followed the [Flutter Style Guide].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/fluttercommunity/plus_plugins/blob/main/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning

### CODE REVIEW

**Comment by miquelbeltran** [comment]
Hey! Due to changes in our contributor guidelines, all PRs should now:

- Have a title that follows the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0-beta.4/) format. e.g. `feat(package_name)!: feature description`
- **DO NOT** modify the CHANGELOG.md or the version in the `pubspec.yaml`. This is going to be an automated process now.

We would ask you kindly to update the PR following these changes.

Thanks!

**Comment by werainkhatri** [comment]
> - Have a title that follows the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0-beta.4/) format. e.g. `feat(package_name)!: feature description`

done ✅

> - **DO NOT** modify the CHANGELOG.md or the version in the `pubspec.yaml`. This is going to be an automated process now.

ya i saw got to know. i haven't modified these files, instead have added the checkbox stating i "haven't" done that.

File: packages/connectivity_plus/connectivity_plus/android/src/main/java/dev/fluttercommunity/plus/connectivity/Connectivity.java [hunk 1]
```diff
@@ -56,12 +60,14 @@ private String getNetworkTypeLegacy() {
     int type = info.getType();
     switch (type) {
       case ConnectivityManager.TYPE_BLUETOOTH:
-        return "bluetooth";
+        return CONNECTIVITY_BLUETOOTH;
```
Comment (miquelbeltran) [inline packages/connectivity_plus/connectivity_plus/android/src/main/java/dev/fluttercommunity/plus/connectivity/Connectivity.java:63]: Thanks for the improvement!

File: packages/connectivity_plus/connectivity_plus_linux/test/connectivity_plus_linux_test.dart [hunk 2]
```diff
@@ -59,6 +59,21 @@ void main() {
...
+          .thenReturn(NetworkManagerConnectivityState.full);
+      when(client.primaryConnectionType).thenReturn('vpn');
+      return client;
+    };
+    expect(
+      linux.checkConnectivity(),
+      completion(equals(ConnectivityResult.vpn)),
+    );
+  });
+
```
Comment (miquelbeltran) [inline packages/connectivity_plus/connectivity_plus_linux/test/connectivity_plus_linux_test.dart:76]: thanks for including tests!

**Review by miquelbeltran — APPROVED** [review]
Thanks for another great contribution!

**Pull request merged**